### ISSUE TITLE
Widgets and Navigation: the main H1 heading must come before any other content in the main region

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#24937]
**Describe the bug**

For semantics and accessibility, all the admin pages have a main H1 heading that identifies "what" the page is about. This main heading must come before any other content within the `role="main"` region of the page.

Over the last years a good amount of work has been made to make all the admin pages have a good heading structure. See for example:
https://make.wordpress.org/core/2017/05/17/cleaner-headings-in-the-admin-screens/
https://make.wordpress.org/core/2015/07/31/headings-in-admin-screens-change-in-wordpress-4-3/
https://make.wordpress.org/core/2015/10/28/headings-hierarchy-changes-in-the-admin-screens/

(note: the only missing part is that in some pages the "Help" and "Screen Options" are actually placed before the main H1: this is not ideal but it can't be changed without a new design that didn't happen so far).

However, I see the new design of the Widgets and Navigation pages goes in a different direction. Right now:
- in the Widgets page the main H1 heading is placed between interactive controls, in an order that doesn't make much sense
- the Navigation page doesn't have any heading

This new design needs to be changed.

It is of fundamental importance that any admin page provides a **visible H1 heading placed before any other content in the main section**. Ideally, the order should be changed to something like:

```
<div id="wpbody" role="main">
	...
	<h1 ....
		<div class="components-navigate-regions">
			...
			[toolbar]
			[content]
			etc.
```

This is even more important considering that this new design is likely going to be used for all the new admin pages in WordPress.

**Screenshots**

New Widgets page headings:

<img width="1440" alt="widgets heading" src="https://user-images.githubusercontent.com/1682452/91730832-911e4500-eba6-11ea-9639-8cd4bb72ea6b.png">

New Navigation page headings (no headings at all):

<img width="1440" alt="navigation headings" src="https://user-images.githubusercontent.com/1682452/91730864-98455300-eba6-11ea-9271-d0f28a8672ff.png">

**Comment by afercia** [comment]
Regarding the Navigation page, a visible H1 heading is also proposed in the latest design on #24875.

**Comment by mapk** [comment]
This has been updated in the recent design work and will be implemented with a proper H1 tag as the first item in the main screen area.

![Legacy Widget i5 - 2](https://user-images.githubusercontent.com/617986/91910329-a73e0b00-ec63-11ea-8bd5-8898b6feaf16.png)

**Comment by draganescu** [comment]
This is solved in the current Widgets implementation as it is according to the design spec.
